features/admin.ai.v1/components/loading-screen.tsx

```diff
@@ -0,0 +1,134 @@
+import { useTranslation } from "react-i18next";
+import { ReactComponent as LoadingPlaceholder }
+    from "../../../modules/theme/src/themes/wso2is/assets/images/illustrations/ai-loading-screen-placeholder.svg";
+import useGetAIBrandingGenerationStatus from "../api/use-get-branding-generation-status";
+import "./loading-screen.scss";
+
+export const LoadingScreen = (): JSX.Element => {
+    const { t } = useTranslation();
+    const [ factIndex, setFactIndex ] = useState(0);
+    const facts: string[] = [
+        t("branding:ai.screens.loading.facts.0"),
+        t("branding:ai.screens.loading.facts.1"),
+        t("branding:ai.screens.loading.facts.2")
+    ];
+
+    const { operationId } = useAIBrandingPreference();
+
+    const { data, isLoading } = useGetAIBrandingGenerationStatus(operationId);
+
+    const statusLabels: Record<string, string> = {
+        branding_generation_completed: t("branding:ai.screens.loading.states.8"),
+        color_palette: t("branding:ai.screens.loading.states.5"),
+        create_branding_theme: t("branding:ai.screens.loading.states.7"),
+        extract_webpage_content: t("branding:ai.screens.loading.states.2"),
+        generate_branding: t("branding:ai.screens.loading.states.4"),
+        render_webpage: t("branding:ai.screens.loading.states.1"),
+        style_properties: t("branding:ai.screens.loading.states.6"),
+        webpage_extraction_completed: t("branding:ai.screens.loading.states.3")
+    };
+
+    const statusProgress: Record<string, number> = {
+        branding_generation_completed: 100,
+        color_palette: 75,
+        create_branding_theme: 97,
+        extract_webpage_content: 25,
+        generate_branding: 50,
+        render_webpage: 10,
+        style_properties: 95,
+        webpage_extraction_completed: 30
+    };
+
+    useEffect(() => {
+        const interval: NodeJS.Timeout = setInterval(() => {
+            setFactIndex((factIndex + 1) % facts.length);
+        }, 8000);
+
+        return () => clearInterval(interval);
+    }, [ factIndex ]);
+
+    const getProgress = () => {
+        if (!data) return 0;
+        // Find the last completed status based on the predefined progress mapping
+        let maxProgress: number = 0;
+
+        Object.entries(data.status).forEach(([ key, value ]: [string, boolean]) => {
+            if (value && statusProgress[key] > maxProgress) {
+                maxProgress = statusProgress[key];
+            }
+        });
+
+        return maxProgress;
+    };
+
+    const getCurrentStatus = () => {
+        if (!data) return t("branding:ai.screens.loading.states.0");
+        let currentStatusLabel: string = "branding:ai.screens.loading.states.0";
+
+        Object.entries(data.status).forEach(([ key, value ]: [string, boolean]) => {
+            if (value && statusLabels[key]) {
+                currentStatusLabel = statusLabels[key];
+            }
+        });
+
+        return t(currentStatusLabel);
+    };
```

Comment:
Without having two sepreate functions we can have a one single useEffect and set the progress and current status to the state variables and use them directly in the component.

Also we do not need `Object.entries` we should be able to directly use `data.status.forEach`

Reply:
The data.status object is not an array, so can't directly use the forEach method on it. improved the logic to use a useEffect for making the progress update smoother with https://github.com/wso2/identity-apps/pull/5961/commits/f29e43a5208f33565cac518ff47efcdf05dc9b5c